src/middleware.ts

```diff
@@ -10,15 +10,18 @@ export function createMiddleware(
 ): ExpressMiddleware {
   return async (req, res, next) => {
     const serverOrigin = `${req.protocol}://${req.get('host')}`
-    const mockedRequest = parseIsomorphicRequest({
-      id: '',
-      method: req.method,
+
+    const mockedRequest = new MockedRequest(
       // Treat all relative URLs as the ones coming from the server.
-      url: new URL(req.url, serverOrigin),
-      headers: new Headers(req.headers as HeadersInit),
-      credentials: 'omit',
-      body: req.body,
-    })
+      new URL(req.url, serverOrigin),
+      {
+        id: '',
+        method: req.method,
+        headers: new Headers(req.headers as HeadersInit),
+        credentials: 'omit',
+        body: req.body,
```

Comment:
Although we keep `body` as pretty much any body type, we should lean towards passing request body as `ArrayBuffer`. In the future (and internally now), request bodies are always represented as buffers as a common ground between different body types. 

```js
import { encodeBuffer } from '@mswjs/interceptors'

body: encodeBuffer(req.body),
```